protocols/shuffling_protocol.go

```diff
@@ -116,6 +118,10 @@ func NewShufflingProtocol(n *onet.TreeNodeInstance) (onet.ProtocolInstance, erro
 			break
 		}
 	}
+
+	// default timeout
+	dsp.Timeout = 10 * time.Minute
```

Comment:
How has this timeout been chosen?

Why are there different timeouts?

Shouldn't it be possible to change the timeout, e.g., with an environment variable?

Reply:
There was no rule to chose the timeout. I just chose enough time for it not to fail in normal query scenarios. As for the timeouts I added one for each different protocol or service. This way we can pass it when calling the protocol or service. You think an environment variable would be better?

Reply:
It depends (as always):
- do you trust the leader to not send a malicious timeout that will keep your resources? (I suppose yes)
- do you want to have each node decide on his own timeouts? (I suppose no)

So with a yes - no answer, I think you're doing the right thing ;) 

But instead of hardcoding the timeout, I'd use https://golang.org/pkg/os/#Getenv to read the `MEDCO_TIMEOUT` environment variable, and, if it is not set, use a globally defined timeout.

Timeouts are always hard to get right (even @bford said "good luck" when I had to do it ;), so having them easily adjustable also from the outside is a plus!

Reply:
@ineiti I followed your suggestion and changed the timeouts 3dcc5b4 
Would this be better?

Reply:
yes, looks good to me.